contracts/ArchetypeLogic.sol

```diff
@@ -141,6 +141,7 @@ library ArchetypeLogic {
     DutchInvite storage invite,
     Discount storage discounts,
     uint256 numTokens,
+    uint256 curSupply,
```

Comment:
I'm pretty sure curSupply does not have to be passed as a function param. 

It can be determined any time by accessing `totalSupply() ` or `_totalMinted`.

We prefer preserving the function signature for sake of backwards compatibility.

Reply:
the scope of the _totalMinted() function is limited to the Archetype contract only, as _totalMinted() is a function of the ERC721AUpgradeable contract and must be inherited from this contract to be able to access it...   

the parameters we need are available in validateMint but not in computePrice, so we need to update the signature of the computePrice function to add the necessary parameter for the linear curve.